book/src/csi-driver-object.md

```diff
@@ -0,0 +1,90 @@
+# CSI CRD: `CSIDriver` Object
+
+## What is the `CSIDriver` object?
+
+The `CSIDriver` Kubernetes API object serves two purposes:
+
+1. Ease driver discovery
+  * If a CSI driver creates a `CSIDriver` object, Kubernetes users can easily discover the CSI Drivers installed on their cluster (simply by issuing `kubectl get CSIDriver`)
+2. Customizing Kubernetes behavior
+  * Kubernetes has a default set of behaviors when dealing with CSI Drivers (for example, it calls the `Attach`/`Detach` operations by default). This object allows CSI drivers to specify how Kubernetes should interact with it.
+
+## What fields does the `CSIDriver` object have?
+
+Here is an example of a v1alpha1 `CSIDriver` object:
+
+```YAML
+apiVersion: csi.storage.k8s.io/v1alpha1
+kind: CSIDriver
+metadata:
+  name: mycsidriver.example.com
+spec:
+  attachRequired: true
+  podInfoOnMountVersion: v1
+```
+
+There are three important fields:
+
+* `name`
+  * This should correspond to the full name of the CSI driver.
+* `attachRequired`
+  * Indicates this CSI volume driver requires an attach operation (because it implements the CSI `ControllerPublishVolume` method), and that Kubernetes should call attach and wait for any attach operation to complete before proceeding to mounting.
+  * If a `CSIDriver` object does not exist for a given CSI Driver, the default is `true` -- meaning attach will be called.
+  * If a `CSIDriver` object exists for a given CSI Driver, but this field is not specified, it defaults to `false` -- meaning attach will not be called.
+  * For more information see [Skip Attach](skip-attach.md).
+* `podInfoOnMountVersion`
+  * Indicates this CSI volume driver requires additional pod information (like pod name, pod UID, etc.) during mount operations.
+  * If value is not specified, pod information will not be passed on mount.
+  * If value is set to a valid version, Kubelet will pass pod information as `volume_context` in CSI `NodePublishVolume` calls.
+  * Supported versions:
+    * Version "v1" will pass the following additional fields in `volume_context`:
+	  * `"csi.storage.k8s.io/pod.name": pod.Name`
+	  * `csi.storage.k8s.io/pod.namespace": pod.Namespace`
+	  * `csi.storage.k8s.io/pod.uid": string(pod.UID)`
+  * For more information see [Pod Info on Mount](pod-info.md).
+
+## What creates the `CSIDriver` object?
+
+CSI drivers do not need to create the `CSIDriver` object directly. Instead they may use the [cluster-driver-registrar](cluster-driver-registrar.md) sidecar container (customizing it as needed with startup parameters) -- when deployed with a CSI driver it automatically creates a `CSIDriver` CR representing the driver.
+
+### Enabling `CSIDriver`
+
+The `CSIDriver` object is available as alpha starting with Kubernetes v1.12. Because it is an alpha feature, it is disabled by default.
+
```

Comment:
Just to make sure: since this is meant to be merged soon (before v1.14 release), we don't want to mention that it's going to be enabled by default in v1.14, correct?

Reply:
Done.